src/mono/wasi/mono-include/driver.h

```diff
@@ -9,6 +9,7 @@ void mono_wasm_load_runtime (const char *unused, int debug_level);
 int mono_wasm_add_assembly (const char *name, const unsigned char *data, unsigned int size);
 MonoAssembly* mono_wasm_assembly_load(const char *name);
 MonoMethod* mono_wasm_assembly_get_entry_point (MonoAssembly *assembly);
```

Comment:
This was my original commit message which kinda got lost in re-arranging the commits!
```
    [wasi] Use the real entrypoint even with async Main

    When the main assembly has a async main, roslyn generates a wrapper
    method which invokes the async main. And in case of wasm that
    async-main method is invoked so it can yield to the host.

    But in case of wasi we invoke the actual entrypoint.
```